cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzurePlatformResources.java

```diff
@@ -121,15 +122,17 @@ public CloudSshKeys sshKeys(CloudCredential cloudCredential, Region region, Map<
     public CloudSecurityGroups securityGroups(CloudCredential cloudCredential, Region region, Map<String, String> filters) {
         AzureClient client = azureClientService.getClient(cloudCredential);
         Map<String, Set<CloudSecurityGroup>> result = new HashMap<>();
-
-        for (NetworkSecurityGroup securityGroup : client.getSecurityGroups().list()) {
-            String actualRegion = securityGroup.region().label();
-            if (regionMatch(actualRegion, region)) {
-                Map<String, Object> properties = new HashMap<>();
-                properties.put("resourceGroupName", securityGroup.resourceGroupName());
-                properties.put("networkInterfaceIds", securityGroup.networkInterfaceIds());
-                CloudSecurityGroup cloudSecurityGroup = new CloudSecurityGroup(securityGroup.name(), securityGroup.id(), properties);
-                result.computeIfAbsent(actualRegion, s -> new HashSet<>()).add(cloudSecurityGroup);
+        PlatformResourceSecurityGroupFilterView filter = new PlatformResourceSecurityGroupFilterView(filters);
+        String groupId = filter.getGroupId();
+        if (groupId != null) {
+            NetworkSecurityGroup networkSecurityGroup = client.getSecurityGroups().getById(groupId);
+            if (networkSecurityGroup == null) {
+                throw new IllegalArgumentException("Nothing found on Azure with id: " + groupId);
```

Comment:
same here. we don't know why the networkSecurityGroup is null, hence this is a state where we don't know what to do. -> IllegalStateException